### PR TITLE
feat(homepage scripts): Add home page scripts

### PR DESCRIPTION
**Pull request by TawKhoury**
…o fb pixel, fix chartbeat

affects: @haaretz/htz-components

ISSUES CLOSED: #1305 

**Table of Contents**

- [Description](#description)
**Update to existing feature**
- [Checklist](#checklist)

**DONE:** 
- [x] Cancel Subscription Popup - 7.4303225 **will be a new task, Create Pop Up**
- [x] Top promotion banner Haaretz - 7.3593804 **ask Rotem: decided that there is no need for this**

- [x] ChromePush - 7.4254397 **already exits, make sure that the same**
- [x] Facebook Instant Article - 7.3824551 **add <meta property="fb:pages" content="109551402519">**
- [x] Facebook Publisher Tag - 7.3331658 **add <meta property="article:publisher" content="https://www.facebook.com/haaretz" />**
- [x] facebook pixel hz - 7.7766984 **check which pixel id to add/ if possible to add to pixels**
- [x] google pixel hz home - 7.7766985 **GoogleReMarketingTag component ready**
- [x] facebook-blink - 7.7768682
- [x] FaceBook Remarketing – commercial - 7.3719429
- [x] new-google-universal - 7.7701129 **check if GoogleAnalyticsInit does the same thing**
- [x] adBlockProbabilities - 7.3826623 **ask Rotem - no need at the moment**
- [x] Facebook Blink מיועד לצביעה של אנונימיים בלבד - שלושת האתרים - 7.7639976 **multiple pixel id problem**
- [x] FIRSTIMPRESSION Direct Haaretz.co.il - 7.7206209 **same as article first impression script**
- [x] Chart Beat - 7.1283191 **check difference with article charbeat**
**cehck why we need 3 charbeat script**
- [x] crazy egg - 7.2844505 **add this**
- [x] IDX-NIELSEN - 7.4198590 **add this**

**To do :**
- [ ] abuse activation script - 7.4207956 **ask tomer/avi how to implement this**
- [ ] adBlock detector - 7.3805210 **ask Tomer/ Rotem/ Gal**
- [ ] ad-block killer - 7.3864935 **ask Tomer**
- [ ] Incognito users block - 7.4389463 **ask how to add PrivateModeUtil**

**Not sure :**
- [ ] outbrain TVR - tracking mode HTZ - 7.7772166 **waiting for Smadar to check if needed**
- [ ] Maavaron script- haaretz+themarker - 7.3426139 **Gal will do this**
 - [ ] default banner rule - 7.3979744 **Rotem / Gal building this**
- [x] Chartbeat2 - 7.4407722  **check difference with article charbeat**
- [x] chartbeat3 - 7.4407723 **check difference with article charbeat**

**Related issue(s):** #1305 

## Description

**Notes:**

## Checklist

Design:
  * [ ] Approved by designer

Usability:
  * [ ] Developer
  * [ ] Project manager

Tests
  
  * [ ] UI snapshot testing 
  * [ ] Unit testing
  * [ ] Integration testing
  * [ ] E2E testing

### CODE REVIEW

File: packages/components/htz-components/src/createDocument.js [hunk 1]
```diff
@@ -87,61 +89,110 @@ const createDocument = ({
...
+             *      SCRIPT ELEMENTS      *
+             * ************************* */}
+            {criticalFont.script}
+
+            {/* ChartBeat scripts should only render on homepage */}
+            {path !== '/' ? null : this.chartbeatConfig()}
+
+            <SEO host={this.props.host} polyFillSrc={polyFillSrc} />
+            {this.renderData()}
+            {process.env.CONNECTION_PRESET === 'stage' ? (
```
Comment (TxHawks) [inline packages/components/htz-components/src/createDocument.js]: Why is this being done based on the connection preset?

File: packages/components/htz-components/src/createApp.js [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
 import { ApolloProvider, } from 'react-apollo';
 import { LevelProvider, } from './components/AutoLevels/LevelContext';
 
 const createApp = (AdditionalComponent = () => null) => class NextApp extends App {
+  componentDidCatch(error, errorInfo) {
+    console.log('CUSTOM ERROR HANDLING !!!!:: ', error);
+    // This is needed to render errors correctly in development / production
+    super.componentDidCatch(error, errorInfo);
+  }
+
```
Comment (TxHawks) [inline packages/components/htz-components/src/createApp.js]: This doesn't seem right? What does this serve?

File: packages/components/htz-components/src/createApp.js [hunk 3]
```diff
@@ -1,9 +1,16 @@
+/* global window document */
```
Comment (TxHawks) [inline packages/components/htz-components/src/createApp.js]: neither `window` not `document` are used here

File: packages/components/htz-components/src/components/Scripts/UniversitiesPush.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
+/* eslint-disable */
+import React, { Component } from 'react';
+import { CookieUtils, } from '@haaretz/htz-user-utils';
+import { appendScript } from '../../utils/scriptTools';
+
+class UniversitiesPush extends Component {
```
Comment (TxHawks) [inline packages/components/htz-components/src/components/Scripts/UniversitiesPush.js]: See as above

File: packages/components/htz-components/src/components/Scripts/IdxNielsen.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    DeviceType: getDeviceType,
+    getTimeStamp,
+  };
+}
+class IdxNielsen extends Component {
+  static propTypes = {
+    shouldRender: PropTypes.bool.isRequired,
+  };
+
+  componentDidMount() {
```
Comment (TxHawks) [inline packages/components/htz-components/src/components/Scripts/IdxNielsen.js]: Why are you creating a class to do this? also, doing this inside `componentDidMount` means that the script is only placed in the dom in the client, and pretty late at that too. It is external to react, so shouldn't be dependant on it.

Just use `dangerouslySetInnerHtml` for this, I think.

@tomerbanin?

File: packages/components/htz-components/src/createDocument.js (hunk 1 above)
Comment (TawKhoury) [reply packages/components/htz-components/src/createDocument.js]: PR #998

**Pull request merged**